coderd/workspaces_test.go

```diff
@@ -427,278 +428,226 @@ func TestWorkspace(t *testing.T) {
 	t.Run("TemplateVersionPreset", func(t *testing.T) {
 		t.Parallel()
 
+		// Test Utility variables
+		templateVersionParameters := []*proto.RichParameter{
+			{Name: "param1", Type: "string", Required: false},
+			{Name: "param2", Type: "string", Required: false},
+			{Name: "param3", Type: "string", Required: false},
+		}
+		presetParameters := []*proto.PresetParameter{
+			{Name: "param1", Value: "value1"},
+			{Name: "param2", Value: "value2"},
+			{Name: "param3", Value: "value3"},
+		}
+		emptyPreset := &proto.Preset{
+			Name: "Empty Preset",
+		}
+		presetWithParameters := &proto.Preset{
+			Name:       "Preset With Parameters",
+			Parameters: presetParameters,
+		}
+
 		testCases := []struct {
 			name                      string
 			presets                   []*proto.Preset
 			templateVersionParameters []*proto.RichParameter
-			selectedPresetIndex       int // -1 if no preset should be used
+			selectedPresetIndex       *int
 		}{
 			{
-				name:                      "No Presets",
+				name:                      "No Presets - No Template Parameters",
 				presets:                   []*proto.Preset{},
-				templateVersionParameters: []*proto.RichParameter{},
-				selectedPresetIndex:       -1,
+				templateVersionParameters: templateVersionParameters,
 			},
 			{
-				name: "Single Preset - No Parameters",
-				presets: []*proto.Preset{{
-					Name: "test",
-				}},
-				templateVersionParameters: []*proto.RichParameter{},
-				selectedPresetIndex:       0,
+				name:                      "No Presets - With Template Parameters",
+				presets:                   []*proto.Preset{},
+				templateVersionParameters: templateVersionParameters,
 			},
 			{
-				name: "Single Preset - With Parameters But No Template Parameters",
-				presets: []*proto.Preset{{
-					Name: "test",
-					Parameters: []*proto.PresetParameter{
-						{Name: "param1", Value: "value1"},
-						{Name: "param2", Value: "value2"},
-					},
-				}},
-				templateVersionParameters: []*proto.RichParameter{},
-				selectedPresetIndex:       0,
+				name:                      "Single Preset - No Preset Parameters But With Template Parameters",
+				presets:                   []*proto.Preset{emptyPreset},
+				templateVersionParameters: templateVersionParameters,
+				selectedPresetIndex:       ptr.Ref(0),
```

Comment:
What about instead just doing something like this?

```suggestion
				selectedPreset:       &emptyPreset,
```